pkg/controller/daemon/daemoncontroller.go

```diff
@@ -716,6 +742,17 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		}
 	}
 
+	dsNeedsSync = dsc.expectations.SatisfiedExpectations(dsKey)
```

Comment:
It's not part of your change, but we don't need to get `dsKey` from `KeyFunc`, right? We could just use the input `key`